internal/handler/user.go

```diff
@@ -0,0 +1,34 @@
+package handler
+
+import (
+	"context"
+	"github.com/dilyara4949/flight-booking-api/internal/domain"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/request"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/response"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"net/http"
+)
+
+type UserService interface {
+	CreateUser(ctx context.Context, signup request.Signup, password string) (domain.User, error)
+	DeleteUser(ctx context.Context, id uuid.UUID) error
+}
+
+func DeleteUserHandler(userService UserService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		userID, err := uuid.Parse(c.Param("userId"))
```

Comment:
You should consider the following flow to properly implement deletion:
1. check whether id from the request (path) is the same as in the token.
2. if yes, then proceed with delete
2. if no, then check whether user have authorization to do deletion (e.g. have an admin  role)
2.1. if has admin role proceed with deletion
3. if the user neither admin nor it's his id, then return "403 Forbidden" response